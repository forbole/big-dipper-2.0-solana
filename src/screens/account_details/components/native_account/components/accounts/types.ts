export type AddressesType = string[];

export type AccountsType = {
  stake: AddressesType;
  nonce: AddressesType;
  vote: AddressesType;
  token: AddressesType;
}
