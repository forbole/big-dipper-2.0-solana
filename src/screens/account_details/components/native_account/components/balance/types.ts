export type BalanceType = {
  native: TokenUnit;
  stake: TokenUnit;
  nonce: TokenUnit;
  vote: TokenUnit;
  total: TokenUnit;
}
