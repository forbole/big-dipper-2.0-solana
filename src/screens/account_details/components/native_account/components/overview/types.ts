export type OverviewType = {
  address: string;
}
