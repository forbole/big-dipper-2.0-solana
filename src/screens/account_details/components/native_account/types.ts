export type OverviewType = {
  address: string;
}

export type AccountType = {
  address: string;
}

export type AccountsType = {
  data: {
    [value:number]: AccountType[];
  };
  count: number;
  loading: boolean;
}

export type NativeAccountState = {
  loading: boolean;
  exists: boolean;
  overview: OverviewType;
}
