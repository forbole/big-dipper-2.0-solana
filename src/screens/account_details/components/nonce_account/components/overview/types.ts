export type OverviewType = {
  address: string;
  balance: number;
  authority: string;
  blockhash: string;
  fee: number;
}
