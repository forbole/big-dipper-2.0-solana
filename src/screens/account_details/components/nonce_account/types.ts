export type OverviewType = {
  address: string;
  authority: string;
  balance: TokenUnit;
  blockhash: string;
  fee: number;
}

export type NonceAccountState = {
  loading: boolean;
  overview: OverviewType;
}
