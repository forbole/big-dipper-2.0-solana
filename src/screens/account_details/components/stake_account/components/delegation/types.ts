export type DelegationType = {
  activated: TokenUnit;
  activating: TokenUnit;
  total: TokenUnit;
}
