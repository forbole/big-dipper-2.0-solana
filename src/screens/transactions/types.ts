export type TransactionsState = {
  loading: boolean;
  exists: boolean;
  items: Transactions[];
}
