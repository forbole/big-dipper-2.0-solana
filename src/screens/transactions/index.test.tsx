/* eslint-disable */
import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import {
  MockTheme, wait,
} from '@tests/utils';
import { TransactionsDocument } from '@graphql/types';
import Transactions from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Layout: (props) => <div id="Layout" {...props} />,
  Box: (props) => <div id="Box" {...props} />,
  LoadAndExist: (props) => <div id="LoadAndExist" {...props} />,
}));

jest.mock('./components', () => ({
  TransactionsList: (props) => <div id="TransactionsList" {...props} />,
}));

const mockTransactionsDocument = jest.fn().mockResolvedValue({
  "data": {
    "transactions": [
      {
        "slot": 125056600,
        "signature": "AxVwePxYYKqGHSwrrpjwptp9km4N91fYtQcHgYvLNRKPEMp6ZNdQJd6wxA4KF3LHRbKS7iZwMWckmnT9YdATE2Y",
        "error": true,
        "block": {
          "timestamp": "2022-03-15T13:13:45"
        },
        "numInstructions": 1
      }
    ]
  }
});

// ==================================
// unit tests
// ==================================
describe('screen: Transactions', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();

    mockClient.setRequestHandler(
      TransactionsDocument,
      mockTransactionsDocument,
    );

    let component;

    renderer.act(() => {
      component = renderer.create(
        <RecoilRoot>
          <ApolloProvider client={mockClient}>
            <MockTheme>
              <Transactions />
            </MockTheme>
          </ApolloProvider>
        </RecoilRoot>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
