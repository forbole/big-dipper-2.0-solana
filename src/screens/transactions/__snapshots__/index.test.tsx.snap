// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Transactions matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Layout"
  navTitle="transactions:transactions"
>
  <div
    exists={true}
    id="LoadAndExist"
    loading={false}
  >
    <div
      className="makeStyles-box"
      id="Box"
    >
      <div
        id="TransactionsList"
        items={
          Array [
            Object {
              "numInstructions": 1,
              "signature": "AxVwePxYYKqGHSwrrpjwptp9km4N91fYtQcHgYvLNRKPEMp6ZNdQJd6wxA4KF3LHRbKS7iZwMWckmnT9YdATE2Y",
              "slot": 125056600,
              "success": true,
              "timestamp": "2022-03-15T13:13:45",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
